### DISCUSSION TITLE
Trouble with a rails app

### DISCUSSION THREAD

**Discussion by joshfester** [piku/piku#326] [Q&A] [answered]
I copied the ENV and Procfile from the example rails app, but running into an issue deploying. Here's the output when I try it:

```
remote: Resolving deltas: 100% (479/479), done.
remote: -----> Creating app 'stocksgalaxy'
remote: warning: remote HEAD refers to nonexistent ref, unable to checkout.
remote:
remote: -----> Deploying app 'stocksgalaxy'
remote: HEAD is now at c72c467 piku config
remote: -----> Ruby Application app detected.
remote: -----> Checking requirements: ['ruby', 'gem', 'bundle']
remote: [None, None, None]
remote: -----> Could not detect runtime!
```

Any help would be greatly appreciated! I'm hoping to switch from Dokku to Piku so that I can use ARM servers

**Comment by rcarmo** [comment, marked as answer]
Well, it would be great if you had Ruby installed - because that's what's the last few lines are telling you :)

**Reply by joshfester** [reply]
That makes a lot of sense 🤓 ...I'm too used to Dokku!

**Comment by chr15m** [comment]
If somebody can tell me the packages required I'll add them to a bootstrap playbook.

**Reply by joshfester** [reply]
Rails has an official Dockerfile generator that uses these (debian) packages by default:
- build-essential 
- curl

Most Rails projects are also using Postgres, which needs these for the pg gem:
- libpq-dev
- postgresql-client

For installing Ruby, the best option by far imo is [asdf](https://asdf-vm.com/). It's extremely simple and Just Works™️

The Rails Dockerfile also does some things with bundle, but that may be beyond the scope here
